switcheroo/custom_keygen.py

```diff
@@ -105,37 +104,37 @@ def generate_private_public_key(cls) -> tuple[bytes, bytes]:
         public_key = key.public_key().export_key(format="OpenSSH")
         return private_key, public_key
 
+    @classmethod
+    def store_private_key(cls, private_key: bytes, private_key_dir: Path):
+        private_key_dir.mkdir(parents=True, exist_ok=True)
+        private_key_path = private_key_dir / KeyGen.PRIVATE_KEY_NAME
+        # If old key is there, delete it
+        private_key_path.unlink(missing_ok=True)
+        private_key_path.touch(mode=0o600)
+        with open(private_key_path, mode="wb") as private_out:
```

Comment:
we are still not using an `opener` - this is better, but you're still doing more work than needed

Reply:
👍